base/path.jl

```diff
@@ -125,6 +125,23 @@ abspath(a::AbstractString, b::AbstractString...) = abspath(joinpath(a,b...))
     end
 end
 
+@windows_only longpath(path::AbstractString) = longpath(utf16(path))
+@windows_only function longpath(path::UTF16String)
+    p = UInt32((sizeof(path)>>2) + 1)
```

Comment:
I just blindly copied the existing `@windows_only realpath`'s
implementation to implement `longpath`, but I am also puzzled by that.

On Mon, Oct 12, 2015 at 11:50 AM, Steven G. Johnson <
notifications@github.com> wrote:

> In base/path.jl
> <https://github.com/JuliaLang/julia/pull/13542#discussion_r41770705>:
>
> > @@ -125,6 +125,23 @@ abspath(a::AbstractString, b::AbstractString...) = abspath(joinpath(a,b...))
> >      end
> >  end
> >
> > +@windows_only longpath(path::AbstractString) = longpath(utf16(path))
> > +@windows_only function longpath(path::UTF16String)
> > +    p = UInt32((sizeof(path)>>2) + 1)
>
> Why are you dividing by 4 (>>2) instead of by 2 (>>1)? Presumably what
> you want is the number of code units, including the null terminator. Maybe
> just UInt32(length(path.data))?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JuliaLang/julia/pull/13542/files#r41770705>.
>